src/react/index.jsx

```diff
@@ -0,0 +1,46 @@
+import React, {useState} from 'react';
+import ReactDOM from 'react-dom';
+import {
+    BrowserRouter as Router,
+    Switch,
+    Route,
+    Link,
+    useHistory
+} from "react-router-dom";
+
+function App() {
+    return (
+    <Router>
+        <div>
+            <Switch>
+                <Route path="/main">
+                    <Main />
+                </Route>
+                <Route path="/page2">
+                    <Page2 />
+                </Route>
+            </Switch>
+        </div>
+    </Router>
+    );
+}
+
+ReactDOM.render(<App />, document.getElementById('app'));
+
+function Main() {
+    const [count, setCount] = useState(0);
+    const history = useHistory();
+    return (
+        <div>
+            <div>My Flask React App!{count}</div>
+            <button onClick={() => setCount(count + 1)}>count up</button>
+            <button onClick={() => history.push("/page2")}>go to other page</button>
+        </div>
+    );
+}
+
+function Page2() {
```

Comment:
Just some info about React: `Main()` and `Page2()` are components in React. They are defined as functions. The `return` value is what they will render. The `return` can use either plain HTML or other React components. Above the `return` is any logic needed for the return, such as keeping track of state, fetching data, calculations, etc.

Reply:
Also I should make this component do a `fetch` out to an API endpoint just to prove we have the front/backend separation we're looking for